vtex/faststore
# Add: Enable Promotions Associated to Campaign Audiences

## What's the purpose of this pull request?

Enable the display of exclusive prices at FastStore for specific clients within a campaign's audience:
https://help.vtex.com/en/tutorial/creating-campaign-audiences--6cnuDZJzIkIeocewAQQK4K

## How it works?

Price for users not associated with a campaign or unidentified customers:
![image](https://github.com/vtex/faststore/assets/67066494/002e7c44-2b92-451e-8505-d63f5ac2bf06)

Price with user associated:

**PDL**
<img width="1272" alt="image" src="https://github.com/vtex/faststore/assets/67066494/37619935-cf74-4fe4-8adf-e757a9ef3184">

**PDP**
<img width="1309" alt="image" src="https://github.com/vtex/faststore/assets/67066494/04b89184-c72a-4312-845b-1c9bf8d04489">

**Search Bar**
<img width="1309" alt="image" src="https://github.com/vtex/faststore/assets/67066494/97217baa-ad91-4ea0-857d-e45665520fa9">


## How to test it?

Test with an user associated to the campaign to get the result informations are inside this document:
https://docs.google.com/document/d/19EMRft5u6qqYmeLDZdoUOeg26x8wUmWJIqH5i9bTSXA/edit?usp=sharing

Create the auth cookie at the FastStore:
document.cookie = `"VtexIdclientAutCookie_storeframework={{cookie_Value}}; path=/; domain=.localhost;"`;

Check the item associated with the promotion:
http://localhost:3000/s?q=58982069&sort=score_desc&page=0


## References

Campaign Created:
https://storeframework.myvtex.com/admin/rnb/#/campaign/c604a1c8-56a3-4ccc-b123-3a3e6feaa6b2?page=1&perPage=20

Promotion Created:
https://storeframework.myvtex.com/admin/promotions/8df54cdc-3b30-490b-b7fa-6b06a6bbff29/

Tutorials related:
https://help.vtex.com/en/tutorial/campaign-promotion--1ChYXhK2AQGuS6wAqS8Ume?&utm_source=autocomplete
https://help.vtex.com/en/tutorial/creating-a-campaign-audience--6cnuDZJzIkIeocewAQQK4K?&utm_source=autocomplete
